guppylang/checker/expr_checker.py

```diff
@@ -266,8 +266,9 @@ def visit_Call(self, node: ast.Call, ty: Type) -> tuple[ast.expr, Subst]:
             node.func = instantiate_poly(node.func, func_ty, inst)
             return with_loc(node, LocalCall(func=node.func, args=args)), return_ty
 
-        if isinstance(func_ty, TupleType) and parse_function_tensor(func_ty):
-            function_elements = parse_function_tensor(func_ty)
+        if isinstance(func_ty, TupleType) and (
+            function_elements := parse_function_tensor(func_ty)
+        ):
```

Comment:
One thing to note: An empty `function_elements` lists also evaluate to `False` which means we can't call empty tuples. But I think that's fine, the expression `()()` isn't that useful :D